src/System.Private.CoreLib/shared/System/Text/ASCIIEncoding.cs

```diff
@@ -58,22 +56,29 @@ internal sealed override void SetDefaultFallbacks()
         public override unsafe int GetByteCount(char[] chars, int index, int count)
         {
             // Validate input parameters
-            if (chars == null)
-                throw new ArgumentNullException(nameof(chars), SR.ArgumentNull_Array);
 
-            if (index < 0 || count < 0)
-                throw new ArgumentOutOfRangeException((index < 0 ? nameof(index) : nameof(count)), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (chars is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.chars, ExceptionResource.ArgumentNull_Array);
```

Comment:
Do ThrowHelpers here make any measurable difference?

Reply:
They did in an earlier iteration of this code. They sufficiently shrank the method body to such an extent that some of the methods were themselves inlined into their callers, even without the aggressive inlining attribute applied. I haven't measured the most recent iteration.